DashAI/back/database/models.py

```diff
@@ -43,21 +43,8 @@ class ModelInstance(Base):
     experiment_id: Mapped[int] = mapped_column(ForeignKey("experiment.id"))
     parameters: Mapped[JSON] = mapped_column(JSON)
     model_name: Mapped[str] = mapped_column(String)
-    run: Mapped["Run"] = relationship(back_populates="run", cascade="all, delete")
-
-
-class Run(Base):
-    __tablename__ = "run"
-    """
-    Table to store all the information about a specific run of a model.
-    """
-    id: Mapped[int] = mapped_column(primary_key=True)
-    model_id: Mapped[int] = mapped_column(ForeignKey("model_instance.id"))
     train_results: Mapped[JSON] = mapped_column(JSON)
     test_results: Mapped[JSON] = mapped_column(JSON)
     validation_restuls: Mapped[JSON] = mapped_column(JSON)
-    weights_path: Mapped[str] = mapped_column(String)
+    artifact_path: Mapped[str] = mapped_column(String)
     trained: Mapped[Boolean] = mapped_column(Boolean)
-    model: Mapped["ModelInstance"] = relationship(
-        back_populates="model_instance", cascade="all, delete"
-    )
```

Comment:
Responding to what you commented for the artifact comments:

I was looking at how mflow implements this. In its most basic form (only with folders) they keep 4 different folders for each run: artifacts, metrics, params and tags. 
Inside artifacts live the serialized model as well as the graphics, inside metrics live all the metrics, params live the parameters (which would be ModelInstance in our case) and tags like the start and end date of execution, user that launched the run, execution environment, etc...

Maybe we could adapt this structure to more structured dash ideas: 

- artifacts to be a json with key the artifact name and value the path.
- metrics leave it as it is
- add tags to the same run
- parameters: merge model instance with this table.

Below is a suggestion of the above (deleting ModelInstance table):

```python
class RunStatus:
    NOTSTARTED = 0
    STARTED = 1
    FINISHED  = 2
    ERROR = 3

class Run(Base):
    __tablename__ = "run"
    """
    Table to store all the information about a specific run of a model.
    """
    id: Mapped[int] = mapped_column(primary_key=True)
    # model and parameters
    model_name: Mapped[str] = mapped_column(String)
    parameters: Mapped[JSON] = mapped_column(JSON)
    # metrics
    train_metrics: Mapped[JSON] = mapped_column(JSON)
    test_metrics: Mapped[JSON] = mapped_column(JSON)
    validation_metrics: Mapped[JSON] = mapped_column(JSON)
    # artifacts
    artifacts: Mapped[str] = mapped_column(JSON)
    # metadata
    run_name: Mapped[str] = mapped_column(String)
    run_description: Mapped[str] = mapped_column(String)
    status: Mapped[Enum] = mapped_column(Enum(RunStatus), nullable=False)
    start_time: Mapped[DateTime] = mapped_column(DateTime)
    end_time: Mapped[DateTime] = mapped_column(DateTime)
    end_time: Mapped[DateTime] = mapped_column(DateTime)
```

We could still discuss this at a meeting, what do you think?

Reply:
This is a good idea, i made some minor fixes (its missing experiment_id, end_time is duplicated, etc). Hope this has enough info for our future orchestrator